packages/math-input/src/store/keypad-reducer.ts

```diff
@@ -0,0 +1,62 @@
+import KeyConfigs from "../data/key-configs";
+import Keys from "../data/keys";
+
+import {defaultKeypadType} from "./shared";
+
+import type {KeypadState} from "./types";
+
+const initialKeypadState = {
+    extraKeys: ["x", "y", Keys.THETA, Keys.PI],
+    keypadType: defaultKeypadType,
+    active: false,
+} as const;
+
+const keypadReducer = function (
+    state = initialKeypadState,
+    action: {
+        type: string;
+    },
```

Comment:
**suggestion:** this needs a better type so that we aren't using `@ts-expect-error` in so many places.